src/en/news/blog/2022/bucket-notifications-with-knative-and-rook-on-minikube2/index.md

```diff
@@ -0,0 +1,330 @@
+    rook_object_store: my-store
+  sessionAffinity: None
+  type: NodePort
+EOF
+```
+
+Well, the truth is that we don't want to access it from the node (the minikube VM), we want to access it from the machine hosting that VM.
+Minikube will help us here, and give us the actual host name we should use:
+
+```bash
+export AWS_URL=$(minikube service --url rook-ceph-rgw-my-store-external -n rook-ceph)
+```
+
+### User Credentials
+
+We get the user credentials and put them into into environment variables used by the aws CLI tool:
+
+```bash
+export AWS_ACCESS_KEY_ID=$(kubectl -n default get secret ceph-notification-bucket -o jsonpath='{.data.AWS_ACCESS_KEY_ID}' | base64 --decode)
+export AWS_SECRET_ACCESS_KEY=$(kubectl -n default get secret ceph-notification-bucket -o jsonpath='{.data.AWS_SECRET_ACCESS_KEY}' | base64 --decode)
+```
+
+And then fetch the generated bucket name:
+
+```bash
+export BUCKET_NAME=$(kubectl get objectbucketclaim ceph-notification-bucket -o jsonpath='{.spec.bucketName}')
+```
+
+### Push the Button
+
+Last, we can create a file, and upload it to Ceph.
+
+```bash
+echo "hello world" > hello.txt
+aws --endpoint-url "$AWS_URL" s3 cp hello.txt s3://"$BUCKET_NAME"
+```
+
+We should now see these events in the "event-display" pod log:
+
+```bash
+kubectl logs -l serving.knative.dev/service=event-display -c display-container --tail=100
+```
+
+At this point, the "event-display" pod should be created and exists as long as notifications are being sent from the RGW.
+
+### Debugging Bucket Notifications
+
+To debug the Ceph Source use:
+
+```bash
+kubectl logs -l eventing.knative.dev/sourceName=my-ceph-source --tail 100
+```
+
+The RGW with:
+
+```bash
+kubectl logs -l app=rook-ceph-rgw -n rook-ceph --tail 100
+```
+
+And the Rook operator with:
+
+```bash
+kubectl logs -l app=rook-ceph-operator -n rook-ceph --tail 100
+```
+
+The Ceph toolbox pod may also be useful for debugging, as it holds the `radosgw-admin` and other tools:
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/rook/rook/release-1.8/deploy/examples/toolbox.yaml
+```
+
+## What's Next?
+
+Looks like we are done? Well, not really.
+
+There are several alternatives to the technologies that we selected above that are worth exploring. So, stay tuned!
+
+### microshift
+
+[microshift](https://github.com/redhat-et/microshift) is a small footprint alternative to minikube (based on Openshift).
+Note that microshift runs directly on the host, and an extra **physical** disk is needed (e.g. attach a USB drive).
+Also note that since micorshift is based on Openshift and not vanilla k8s, there are some different steps in installaing the other components.
+
+### CRC
+
+[CodeReadty Containers](https://github.com/code-ready/crc) is another, VM based, small footprint, Openshift.
+
+### Cloudevents Endpoint
+
+In the next Rook version there will be a popssibility to send notifications as Cloudevents directly to the Knative channel, without the "Ceph Sourcee" adapter
```

Comment:
In the next Rook version, there will be a possibility to send notifications as Cloudevents directly to the Knative channel, without the "Ceph Source" adapter